src/main/scala/mesosphere/marathon/core/deployment/impl/StartingBehavior.scala

```diff
@@ -47,16 +43,24 @@ trait StartingBehavior extends ReadinessBehavior with StrictLogging { this: Acto
     case InstanceChanged(id, `version`, `pathId`, condition: Condition, instance) if condition.isTerminal || instance.isReservedTerminal =>
       logger.warn(s"New instance [$id] failed during app ${runSpec.id.toString} scaling, queueing another instance")
       instanceTerminated(id)
-      launchQueue.add(runSpec, 1).pipeTo(self)
+      scheduler.schedule(runSpec, 1).pipeTo(self)
 
     case Sync => async {
-      val instances = await(instanceTracker.specInstances(runSpec.id))
+      val instances = await(scheduler.getInstances(runSpec.id))
       val actualSize = instances.count { i => i.isActive || i.isScheduled }
       val instancesToStartNow = Math.max(scaleTo - actualSize, 0)
       logger.debug(s"Sync start instancesToStartNow=$instancesToStartNow appId=${runSpec.id}")
       if (instancesToStartNow > 0) {
         logger.info(s"Reconciling app ${runSpec.id} scaling: queuing $instancesToStartNow new instances")
-        await(launchQueue.add(runSpec, instancesToStartNow))
+        // Reschedule stopped resident instances first.
+        val existingReservedStoppedInstances = instances
```

Comment:
I disagree. This was my main point against the interface. `Scheduler.schedule` is a `create` operation. `Scheduler.reschedule` is an `update` operation. The orchestrator should explicitly call either one. The `LaunchQueue.Add` would hide the difference. That's why I really do not want to expose that part.

Reply:
oh I see your point... but it would be hard to do until we have event loop because all the places where you do this kind of reschedule will have to be synchronized (you can't have two threads doing that on the same runspec).

Reply:
We always have only one orchestrator running these commands. Also, I used the `WorkQueue` to synchronize the calls. @zen-dog did not like it so we are figuring something else out.